### PR TITLE
Private image support

### PR DESCRIPTION
**Pull request by mikhail-khodorovskiy**
allows using secret manager secrets to authenticated to private docker registry which allows to use datalab private image from the registry for extra libraries, native components, etc.
Issue:  the secrets are splatted into the startup script so console users who have permissions to see instances metadata will see the credentials.

### CODE REVIEW

**Comment by bharathkkb** [comment]
@mikhail-khodorovskiy 
Thanks for the contribution! Looks like the docs were not generated. Could you run `make docker_generate_docs`. More info [here](https://github.com/terraform-google-modules/terraform-google-datalab/blob/master/CONTRIBUTING.md)

File: modules/instance/variables.tf [hunk 1]
```diff
@@ -143,3 +143,13 @@ variable "enable_secure_boot" {
...
+
+variable "private_datalab_registry_info" {
+  type = object({
+    secret_project_id   = string
+    user_secret_id      = string
+    password_secret_id  = string
+    docker_registry_url = string
+  })
+  default = null
+}
```
Comment (bharathkkb) [inline modules/instance/variables.tf:155]: nit : newline

File: modules/template_files/variables.tf [hunk 2]
```diff
@@ -104,3 +104,12 @@ variable "append_to_startup_script" {
...
 }
+
+variable "private_datalab_registry_info" {
+  type = object({
+    secret_project_id   = string
+    user_secret_id      = string
+    password_secret_id  = string
+    docker_registry_url = string
+  })
+}
```
Comment (bharathkkb) [inline modules/template_files/variables.tf:115]: nit: new line

**Comment by g-awmalik** [comment]
Closing since this is stale. Please reopen if necessary.

**Pull request closed without merging**